initiator.go

```diff
@@ -27,67 +29,104 @@ func RunDKG(init *Init) ([]*Result, error) {
 	return results, err
 }
 
-func RunReshare(
-	validatorPK []byte,
-	withdrawalCredentials []byte,
-	fork [4]byte,
-	signedReshare *SignedReshare,
-	proofs map[*Operator]SignedProof,
-	client eip1271.ETHClient,
-) ([]*Result, error) {
-	id := NewID()
+func RunReshare(signedReshare *SignedReshare) ([][]*Result, error) {
+	id, err := GetReqIDfromMsg(signedReshare)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get reqID: %w", err)
+	}
 
-	var results []*Result
+	var results [][]*Result
 	/*
-		DKG ceremony ...
+		reshare ceremonies ...
 	*/
-	_, _, _, err := ValidateResults(
-		signedReshare.Reshare.NewOperators,
-		withdrawalCredentials,
-		validatorPK,
-		fork,
-		signedReshare.Reshare.Owner,
-		signedReshare.Reshare.Nonce,
-		id,
-		results)
-	return results, err
+	for i, reshareMsg := range signedReshare.Messages {
+		_, _, _, err := ValidateResults(
+			reshareMsg.Reshare.NewOperators,
+			reshareMsg.Reshare.WithdrawalCredentials,
+			reshareMsg.Reshare.ValidatorPubKey,
+			reshareMsg.Reshare.Fork,
+			reshareMsg.Reshare.Owner,
+			reshareMsg.Reshare.Nonce,
+			id,
+			results[i])
```

Comment:
As I see @pavelkrolevets updated this in his fork and used that in the pectra PR in ssv-dkg, I think it would be better to create a PR on his side about this. What do you think?